### PR TITLE
elements, filters and elements on elements

### PR DESCRIPTION
**Pull request by alejajessi**
image

### CODE REVIEW

File: src/page/product-list.page.ts [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+    await this.findByProduct(productName).$('a.ajax_add_to_cart_button').click();
+
   }
 
-  public async selectProductTShirt(): Promise<void> {
-    await this.tShirtProduct.click();
+  private findByProduct (productName:string): ElementFinder {
+
+    return this.products.filter((elemento:ElementFinder) =>
+    elemento.$('.product-name').getText().then((texto:string) =>
```
Comment (leonleo997) [inline src/page/product-list.page.ts:27]: indent your code

File: src/page/product-list.page.ts [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
+  private products: ElementArrayFinder;
+  private addProduct: ElementFinder;
+  private timeOut: number = 7000;
 
-  constructor () {
+  constructor() {
+
+    this.products = $$('.product-container');
+    this.addProduct = $('div[class="button-container"] > a[title="Add to cart"]');
+
```
Comment (leonleo997) [inline src/page/product-list.page.ts:12]: delete empty line

File: src/page/product-list.page.ts [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
 
 export class ProductListPage {
-  private tShirtProduct: ElementFinder;
+  private products: ElementArrayFinder;
+  private addProduct: ElementFinder;
+  private timeOut: number = 7000;
 
-  constructor () {
+  constructor() {
+
```
Comment (leonleo997) [inline src/page/product-list.page.ts:9]: delete empty line

File: src/page/product-list.page.ts [hunk 4]
```diff
@@ -1,15 +1,26 @@
...
 export class ProductListPage {
-  private tShirtProduct: ElementFinder;
+  private products: ElementArrayFinder;
+  private addProduct: ElementFinder;
+  private timeOut: number = 7000;
 
-  constructor () {
+  constructor() {
+    this.products = $$('.product-container');
+    this.addProduct = $('div[class="button-container"] > a[title="Add to cart"]');
```
Comment (holgiosalos) [inline src/page/product-list.page.ts:10]: I think that this element is not required

File: src/page/product-list.page.ts [hunk 5]
```diff
@@ -1,15 +1,26 @@
...
+  }
 
-    this.tShirtProduct = $('div[class="button-container"] > a[title="Add to cart"]');
+  public async selectProduct(productName: string): Promise<void> {
+    await browser.wait(ExpectedConditions.visibilityOf(this.findByProduct(productName)),
+                       this.timeOut);
+    await this.addProduct.click();
+    await this.findByProduct(productName).$('a.ajax_add_to_cart_button').click();
```
Comment (holgiosalos) [inline src/page/product-list.page.ts:17]: since here you are clicking here the add to cart button, isn't it?

**Pull request merged**